### PR TITLE
refactor: add 'npm run bootstrap', 'npm run build'

### PR DESCRIPTION
**Pull request by codebytere**
This PR makes our script commands simpler by adding `bootstrap:dev`, `build:release`  and `build:dev` options and updating docs to reflect that most developers working on Electron should be bootstrapping for dev.

Also adds a script for `check-tls`.

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: docs/development/build-instructions-linux.md [hunk 1]
```diff
@@ -73,7 +73,14 @@ Downloading certain files can take a long time. Notice that we are using
 
 ```sh
 $ cd electron
-$ ./script/bootstrap.py --verbose
+$ npm run bootstrap
+```
+
+You can also bootstrap for debugging:
```
Comment (MarshallOfSound) [inline docs/development/build-instructions-linux.md]: for debug builds

File: docs/development/build-instructions-linux.md [hunk 2]
```diff
@@ -73,7 +73,14 @@ Downloading certain files can take a long time. Notice that we are using
...
+
+```sh
+$ cd electron
+$ npm run bootstrap-dev
```
Comment (MarshallOfSound) [inline docs/development/build-instructions-linux.md]: I'm a fan of the `bootstrap`, `bootstrap:debug`, `bootstrap:release` style of doing _sub_ npm scripts, thoughts?

Comment (codebytere) [reply docs/development/build-instructions-linux.md]: oh i like this too, i'll change it

File: docs/development/build-instructions-osx.md [hunk 3]
```diff
@@ -86,13 +93,17 @@ $ ./script/build.py --compdb
 Build both `Release` and `Debug` targets:
 
 ```sh
-$ ./script/build.py
+$ npm run build
 ```
 
-You can also only build the `Debug` target:
+You can also only build the `Debug` target or the `Release` target:
```
Comment (ckerr) [inline docs/development/build-instructions-osx.md]: You can also build either the `Debug` or `Release` target on its own:

File: docs/development/build-instructions-windows.md [hunk 4]
```diff
@@ -39,21 +39,32 @@ there is no Visual Studio project generated.
...
-Build both Release and Debug targets:
+Build both `Release` and `Debug` targets:
 
 ```powershell
-$ python script\build.py
+$ npm run build
 ```
 
-You can also only build the Debug target:
+You can also only build the `Debug` target or the `Release` target:
```
Comment (ckerr) [inline docs/development/build-instructions-windows.md]: You can also build either the `Debug` or `Release` target on its own:

File: docs/development/build-instructions-linux.md [hunk 5]
```diff
@@ -73,7 +73,14 @@ Downloading certain files can take a long time. Notice that we are using
 
 ```sh
 $ cd electron
-$ ./script/bootstrap.py --verbose
+$ npm run bootstrap
+```
+
+You can also bootstrap for debug builds:
```
Comment (ckerr) [inline docs/development/build-instructions-linux.md]: Moot if the above suggestion is taken, but this should be `developer` not `debug`

File: docs/development/build-instructions-linux.md [hunk 6]
```diff
@@ -73,7 +73,14 @@ Downloading certain files can take a long time. Notice that we are using
 
 ```sh
 $ cd electron
-$ ./script/bootstrap.py --verbose
+$ npm run bootstrap
```
Comment (ckerr) [inline docs/development/build-instructions-linux.md]: I think we should be clearer on what leaving out the `:dev` sub does.

Suggestions welcomed on this, but as a starter:

##

To bootstrap for a static, non-developer build, run:

```sh
$ cd electron
$ npm run bootstrap
```

Or to bootstrap for a development session that builds faster by not statically linking:

```sh
$ cd electron
$ npm run boostrap:dev
```

File: docs/development/build-instructions-osx.md [hunk 7]
```diff
@@ -71,7 +71,14 @@ to build Electron so there is no Xcode project generated.
...
 $ cd electron
-$ ./script/bootstrap.py -v
+$ npm run bootstrap
+```
+
+You can also bootstrap for debug builds:
+
+```sh
+$ cd electron
+$ npm run bootstrap:dev
```
Comment (ckerr) [inline docs/development/build-instructions-osx.md]: Same suggestions as above for better explanation of dev and non-dev meaning, and also misuse of `debug` here

File: docs/development/build-instructions-windows.md [hunk 8]
```diff
@@ -39,21 +39,32 @@ there is no Visual Studio project generated.
...
 $ cd electron
-$ python script\bootstrap.py -v
+$ npm run bootstrap
+```
+
+You can also bootstrap for debug builds:
+
+```powershell
+$ cd electron
+$ npm run bootstrap:dev
```
Comment (ckerr) [inline docs/development/build-instructions-windows.md]: Same suggestion as above for explaining `developer` and non-developer bootstrapping, and for not referring to debug bootstraps

File: docs/development/build-instructions-linux.md [hunk 9]
```diff
@@ -128,10 +135,14 @@ This will put a working distribution with much smaller file sizes in
 the `dist` directory. After running the `create-dist.py` script, you
 may want to remove the 1.3+ gigabyte binary which is still in `out/R`.
 
-You can also build the `Debug` target only:
+You can also build the `Debug` target only or the `Release` target only:
```
Comment (ckerr) [inline docs/development/build-instructions-linux.md]: You can also build either the `Debug` or `Release` target on its own:

File: docs/development/build-instructions-linux.md (hunk 6 above)
Comment (codebytere) [reply docs/development/build-instructions-linux.md]: you're definitely correct, this is far better!

**Comment by ckerr** [comment]
(Changing the PR's semantic message from `doc` to `refactor` because this isn't a docfix -- the docs are being changed to describe the new `npm run` alias. Without the refactor, the doc changes wouldn't be necessary)

**Pull request merged**